js/README.md

```diff
@@ -1,3 +1,88 @@
 # JS libraries
 
 TODO: split in a separate repo with npm `package.json`
+
+---
+## Descriptions
+JavaScript libraries based on the ETH implementation of the protocol.
+
+## Channel
+
+### Channel States:
+Unknown: the channel does not exist yet
+Active: the channel exists, has a deposit, and it's within the valid period
+Expired: the channel exists, but it's no longer valid
+[(for more information)](https://github.com/AdExNetwork/adex-protocol#ocean-based-unidirectional-trust-less-payment-channel-outpace)
+
+### Constructor arguments
+
+- `creator` - Address of the creator of the channel
+- `tokenAddress` - Address of the token that will be used for the channel
+- `tokenAmount` - Total monetary deposit of `tokenAddress`
+- `validUntil` - the date until this channel is valid; this is also the period within the publishers can withdraw, so it should be longer than the actual specified campaign length (e.g. 3x longer)
+- `validators` -an array of all the validators who are responsible for signing a new state; one of them should represent the advertiser, and the other - the publisher(s)
+- `spec` - describes all the campaign criteria: e.g. buy as many impressions as possible, the maximum price they're willing to pay for impressions, and campaign duration; this is stored as arbitrary bytes (32); in the platform, we encode the criteria directly in there, but it can be used to reference a JSON descriptor stored on IPFS.
```

Comment:
"in the platform, we encode the criteria directly in there, but it can be used to reference a JSON descriptor stored on IPFS." - no longer accurate - replace by "This is a hash of the JSON of the campaignSpec"